src/redis-check-rdb.c

```diff
@@ -303,9 +303,9 @@ int redis_check_rdb(char *rdbfilename, FILE *fp) {
             robj *o = rdbLoadCheckModuleValue(&rdb,name);
             decrRefCount(o);
             continue; /* Read type again. */
-        } else if (type == RDB_OPCODE_FUNCTION) {
+        } else if (type == RDB_OPCODE_FUNCTION || type == RDB_OPCODE_FUNCTION2) {
```

Comment:
i now realize we need to also update `rdb_check_doing_string` and set `rdbstate.doing` here.

Reply:
Fixed